components/renku_data_services/namespace/blueprints.py

```diff
@@ -236,3 +236,29 @@ async def _get_namespace(_: Request, user: base_models.APIUser, slug: Slug) -> J
             )
 
         return "/namespaces/<slug:renku_slug>", ["GET"], _get_namespace
+
+    def get_namespace_second_level(self) -> BlueprintFactoryResponse:
+        """Get user or group namespace by slug."""
+
+        @authenticate(self.authenticator)
+        async def _get_namespace_second_level(
+            _: Request, user: base_models.APIUser, first_slug: Slug, second_slug: Slug
+        ) -> JSONResponse:
+            path = ProjectPath.from_strings(first_slug.value, second_slug.value)
```

Comment:
question: can this only be a project? it could be a data connector, no?

Reply:
It can only be a project. Because "namespace" or "owner" is something that can hold other things. So only groups, users or projects are namespaces. And we only have 1 namespace that is at this second level. That is `user/project` or `group/project`. We do have paths like `group/dc` but those cannot be namespaces or owners - i.e. you cannot put other things inside `group/dc`. And `get_namespace_by_path` used in the handler can only return group, user or project namespaces. It will filter out anything with data connectors.